sparkle/output/configuration_output.py

```diff
@@ -0,0 +1,243 @@
+                 configurator: Configurator, instance_set_train: InstanceSet,
+                 instance_set_test: InstanceSet,
+                 penalty_multiplier: int, output: Path) -> None:
+        """Initialize Configurator Output class.
+
+        Args:
+            path: Path to configuration output directory
+            solver: Solver object
+            configurator: The configurator that was used
+            instance_set_train: Instance set used for training
+            instance_set_test: Instance set used for testing
+            penalty_multiplier: penalty multiplier that is applied to the par performance
+                [To be Removed]
+            output: Path to the output directory
+        """
+        self.solver = solver
+        self.configurator = configurator
+        self.instance_set_train = instance_set_train
+        self.instance_set_test = instance_set_test
+        self.penalty_multiplier = penalty_multiplier
+        self.directory = path
+
+        if output.is_dir():
+            self.output = output / "configuration.json"
+        else:
+            self.output = output
+
+        # TODO: Fix this spaghetti code to find the scenario file
+        solver_dir_name = path.name
+        scenario_dir = path / "outdir_train_configuration" / \
+            f"{solver_dir_name}_scenario" / "state-run0"
+        if scenario_dir.is_dir():
+            scenario_file = scenario_dir / "scenario.txt"
+            if not scenario_file.is_file():
+                print("Can't find scenario file")
+                return
+        else:
+            print("Can't find scenario file")
+            return
+
+        scenario_dir = scenario_dir.parent
+        # Sets scenario on configurator object
+        self.configurator.scenario = \
+            ConfigurationScenario.from_file(scenario_file, self.solver,
+                                            self.instance_set_train,
+                                            scenario_dir)
+        self.configurator.scenario._set_paths(self.configurator.output_path)
+
+        # Retrieve all configurations
+        config_path = path / "validation" / "configurations.csv"
+        self.configurations = self.get_configurations(config_path)
+
+        # Retrieves validation results and best configuration
+        self.training, self.best_config =\
+            self.get_validation_data(self.instance_set_train)
+
+        # Retrieve test validation results if they exist
+        if self.instance_set_test is not None:
+            self.test, _ = self.get_validation_data(self.instance_set_test)
+
+    def get_configurations(self: ConfigurationOutput, config_path: Path) -> list[dict]:
+        """Read all configurations and transform them to dictionaries."""
+        configs = []
+
+        # TODO: Should we check if the configurations are unique?
+        # Check if the path exists and is a file
+        if config_path.exists() and config_path.is_file():
```

Comment:
Do check if the configurations are unique, you don't have to check in this file that everything exists